unlock-app/src/components/interface/connect/ConnectedWallet.tsx

```diff
@@ -27,6 +28,8 @@ export const ConnectedWallet = ({ onNext }: ConnectedWalletProps) => {
 
   const onSignOut = useCallback(async () => {
     setIsDisconnecting(true)
+    // Next Sign Out should be manually called here before other
```

Comment:
This is bad. Can you _only_ call that if this is a WAAS?
We should really isolate the logic *inside* `signOut` or `deAuthenticate` (and we should really combine these infact...)

Reply:
Components in the UI should not have care/worry about whether a wallet is connected but the token is missing or the either way around. The component should just know if a user is (fully) authenticated OR not.